memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -220,8 +220,9 @@ final case class MutableHistogram(buckets: HistogramBuckets, values: Array[Doubl
    * If the scheme is different, then an approximation is used so that the resulting histogram has
    * an approximate sum of the individual distributions, with the original scheme.  Modifies itself.
    */
-  final def addNoCorrection(other: HistogramWithBuckets): Unit =
-    if (buckets == other.buckets) {
+  final def addNoCorrection(other: HistogramWithBuckets): Unit = {
+    // Allow addition when type of bucket is different
+    if (buckets.allBucketTops.sameElements(other.buckets.allBucketTops)) {
```

Comment:
Sorry but this is problematic.   
1) The bucket tops are not the only thing that matters.  For certain schemes such as DDS, other factors might be important like other parameters, and so this equivalency test is not right.
2) It breaks the abstraction given to buckets to compare themselves.  Comparing all the elements for geometric buckets might not perform as well.

What I suggest is instead to add a method `similarForMath` to each bucket which returns true for this purpose, if the buckets are similar enough for additions.   In that method, it can then switch on different types of `other` and do what is optimal and correct.  This would let subtypes decide when they can be added.